snoop/data/analyzers/archives.py

```diff
@@ -195,21 +193,28 @@ def unarchive(blob):
 
 def archive_walk(path):
     """Generates simple dicts with archive listing for the archive. """
-
-    for thing in path.iterdir():
-        if thing.is_dir():
-            yield {
-                'type': 'directory',
-                'name': thing.name,
-                'children': sorted(
-                    list(archive_walk(thing)),
-                    key=lambda c: c['name'],
-                ),
+    res = []
+    for root, dirs, files in os.walk(path, topdown=True):
+        children = []
+        for d in dirs:
+            dir_info = {
+                'type' : 'directory',
+                'name' :  os.path.relpath(d),
+                'children' : [],
             }
-
-        else:
-            yield {
-                'type': 'file',
-                'name': thing.name,
-                'blob_pk': models.Blob.create_from_file(thing).pk,
+            children.append(dir_info)
+        for f in files:
+            file_info = {
+                'type' : 'file',
+                'name' : os.path.relpath(f),
+                'blob_pk' : models.Blob.create_from_file(Path(os.path.join(root, f))).pk,
             }
+            children.append(file_info)
+        root_info = {
+            'type' : 'directory',
+            'name' : os.path.relpath(root),
```

Comment:
same as above: this should be the root name of object (for the root directory it's probably `''` the empty string)